### ISSUE TITLE
Dotted dict keys are misclassified as non-differentiable in _merge_path

### ISSUE THREAD

**Issue by dionhaefner** [pasteurlabs/tesseract-jax#251]
## Description

A `dict[str, Differentiable[...]]` field whose keys contain dots — the natural shape for a state dict, e.g. `encoder.layer.0.weight` — is reported differentiable and applies fine, but a gradient through it either fails with a message that points nowhere near the cause or is silently classified as non-differentiable.

The dict-key gradient path is spelled `params.{<key>}` on the wire. `_merge_path` in [`tesseract_jax/tree_util.py`](https://github.com/pasteurlabs/tesseract-jax/blob/main/tesseract_jax/tree_util.py) splits **both** the concrete path and the template on every `.` before comparing them:

```python
explicit_parts = explicit_path.split(".")
...
template_parts = array_path.split(".")
if len(template_parts) != len(explicit_parts):
    continue
```

A concrete path built from a dotted key has more parts than its single-`{}` template, so the length check never matches and `_merge_path` returns `(explicit_path, None)`. The `None` matched-template is what gets read downstream as "this field is not differentiable", which is why the surfaced error talks about a tangent rather than the key.

```python
_merge_path('params.plain',           ['params.{}'])  ->  ('params.{plain}', 'params.{}')  # ok
_merge_path('params.a/b',             ['params.{}'])  ->  ('params.{a/b}',   'params.{}')  # ok (slashes are fine)
_merge_path('params.layer.0.weight',  ['params.{}'])  ->  ('params.layer.0.weight', None)  # dotted key: no match
```

This is the tesseract-jax side of pasteurlabs/tesseract-core#705. The tesseract-core half — widening the schema pattern to `\{[^.]+\}` so slashed/colon'd keys resolve — landed in pasteurlabs/tesseract-core#707. Dotted keys were deliberately left refused there because they need a brace-aware split, which is the part that lives here.

The fix is to make the split in `_merge_path` brace-aware, so a `{...}` segment is treated as one part regardless of the dots inside it, rather than splitting on every `.`.

## Steps to reproduce

```python
import jax
import numpy as np
from tesseract_core import Tesseract
from tesseract_jax import apply_tesseract

# A Tesseract with an input field:
#   params: dict[str, Differentiable[Array[(3,), Float32]]]

for key in ("plain", "a/b", "layer.0.weight"):
    inputs = {"params": {key: np.ones(3, dtype=np.float32)}}
    try:
        g = jax.grad(lambda x: apply_tesseract(t, x)["result"].sum())(inputs)
        print(f"{key!r:20} OK")
    except Exception as e:
        print(f"{key!r:20} {type(e).__name__}: {e}")
```

## Logs

```bash
'plain'              OK
'a/b'                jsonschema error: String should match pattern   # fixed by tesseract-core#707
'layer.0.weight'     Non-symbolic-zero tangent provided for non-differentiable input 'params.layer.0.weight'
```

(The `a/b` line now resolves once the tesseract-core#707 pattern fix is picked up; the dotted line is the remaining tesseract-jax bug.)

## OS

Linux, Mac

## Tesseract + Tesseract-JAX version

tesseract-core >= 1.12.0 (with pasteurlabs/tesseract-core#707), tesseract-jax latest `main`
